addons/im_livechat/static/src/legacy/public_livechat_chatbot.js

```diff
@@ -0,0 +1,882 @@
+odoo.define('im_livechat.legacy.im_livechat_chatbot.im_livechat', function (require) {
+"use strict";
+
+const core = require('web.core');
+const localStorage = require('web.local_storage');
+const session = require('web.session');
+const time = require('web.time');
+const utils = require('web.utils');
+
+const LivechatButton = require('im_livechat.legacy.im_livechat.im_livechat').LivechatButton;
+
+const _t = core._t;
+const QWeb = core.qweb;
+
+/**
+ * Override of the LivechatButton to include chatbot capabilities.
+ * Main changes / hooking points are:
+ * - Show a custom welcome message that is in fact the first message of the chatbot script
+ * - When messages are rendered, add click handles to chatbot options
+ * - When the user picks an option or answers to the chatbot, display a "chatbot is typing..."
+ *   message for a couple seconds and then trigger the next step of the script
+ */
+LivechatButton.include({
+    init: function () {
```

Comment:
Since `willStart()` uses the short method syntax there is no reason for the other methods not to do so as well, being sync is not an impediment.